web/ui/react-app/src/hooks/useFetch.ts

```diff
@@ -86,33 +57,61 @@ export const useFetchReadyInterval = (pathPrefix: string, options?: RequestInit)
   const [walReplayStatus, setWALReplayStatus] = useState<WALReplayStatus>({} as any);
 
   useEffect(() => {
-    const interval = setInterval(async () => {
-      try {
-        let res = await fetch(`${pathPrefix}/-/ready`, { cache: 'no-store', credentials: 'same-origin', ...options });
-        if (res.status === 200) {
-          setReady(true);
-          clearInterval(interval);
-          return;
-        }
-        if (res.status !== 503) {
-          setIsUnexpected(true);
-          setWALReplayStatus({ data: { last: 0, first: 0 } } as any);
-        } else {
-          setIsUnexpected(false);
+    if (wasReady) {
+      setReady(true);
+    } else {
+      // This helps avoid a memory leak.
+      let mounted = true;
 
-          res = await fetch(`${pathPrefix}/${API_PATH}/status/walreplay`, { cache: 'no-store', credentials: 'same-origin' });
-          if (res.ok) {
-            const data = (await res.json()) as WALReplayStatus;
-            setWALReplayStatus(data);
+      const fetchStatus = async () => {
+        try {
+          let res = await fetch(`${pathPrefix}/-/ready`, { cache: 'no-store', credentials: 'same-origin', ...options });
+          if (res.status === 200) {
+            if (mounted) {
+              setReady(true);
+            }
+            wasReady = true;
+            clearInterval(interval);
           }
+          if (res.status !== 503) {
```

Comment:
Question: doesn't this set `isUnexpected` to `true` whenever we get a `200` status code? Should it be an `else if` and not just an `if`?

Reply:
Oh yeah probably.